### PR TITLE
docs: Import certificate with ingress deployment

### PR DESCRIPTION
**Pull request by balteravishay**
**Reason for Change**:

* This PR documents how to load tls certificates, which are used by ingress for HTTPS, by using the ingress controller's deployment, instead of the application's. In this way the application's container does not have access to the volume containing the certificate.
* Improve the links to samples in the main readme.
* Consolidate samples folder under docs.

**Requirements**

- [v] squashed commits
- [v] included documentation
- [na] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Please answer the following questions with yes/no**:

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
no

**Special Notes for Reviewers**:

### CODE REVIEW

File: docs/ingress-tls.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
+# Using Secrets Store CSI and Azure Key Vault Provider to Enable NGINX Ingress Controller with TLS
+
+This guide demonstrates steps required to setup Secrets Store CSI driver and Azure Key Vault Provider to enable applications to work with NGINX Ingress Controller with TLS certificates stored in Key Vault.
+For more information on securing an Ingress with TLS, refer to [this guide](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls)
+
+Importing the ingress TLS certificate to the cluster can be included in the following deployments:
+
+* Ingress Controller - Ingress deployment is modified to declare and mount the csi provider volume. The secret is imported when Ingress pods are created and the application's pods have no access to the TLS certificate. This scenario fits teams where Ingress and its related certificates are managed centrally by an IT unit.
```
Comment (aramase) [inline docs/ingress-tls.md:8]: The CSI driver creates the Kubernetes secrets in the same namespace as the application pod requesting volume mount. So in this case the ingress resource will also need to be created in the same namespace as ingress controller as Kubernetes doesn't allow referencing secret from different namespace. Is this something users currently do?

File: docs/ingress-tls.md [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+```
+
+### Import the TLS certificate to Azure Key Vault
+
+Convert the .crt certificate to pfx format and import it to Azure Key Vault. For example:
+
+```bash
+export PASSOWRD="[YOUR PASSWORD]"
+export AKV_NAME="[YOUR AKV NAME]"
+openssl pkcs12 -export -in ingress-tls.crt -inkey ingress-tls.key  -out $CERT_NAME.pfx -passout pass:$PASSWORD
```
Comment (aramase) [inline docs/ingress-tls.md:28]: The AKV provider for CSI driver currently converts PFX to PEM without a password. If the content is password protected then the conversion will fail.

Ref: https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/pkg/provider/provider.go#L434

File: docs/ingress-tls.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+az keyvault certificate import --vault-name $AKV_NAME -n $CERT_NAME -f $CERT_NAME.pfx --password $PASSWORD
+```
+
+## Setup Cluster Prerequisites
+
+### Deploy Secrets-store CSI and the Azure Key Vault Provider
+
+[Follow this tutorial](https://github.com/Azure/secrets-store-csi-driver-provider-azure#install-the-secrets-store-csi-driver-and-the-azure-keyvault-provider) to setup the secret-store csi and azure key vault provider in the cluster.
```
Comment (aramase) [inline docs/ingress-tls.md:37]:
```suggestion
[Follow this tutorial](https://github.com/Azure/secrets-store-csi-driver-provider-azure#install-the-secrets-store-csi-driver-and-the-azure-keyvault-provider) to setup the secret store csi driver and azure key vault provider in the cluster.
```

File: docs/ingress-tls.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+      - name: secrets-store-inline
+        mountPath: "/mnt/secrets-store"
+        readOnly: true
+EOF
+```
+
+If not using [service principal mode](./service-principal-mode.md), remove the following snippet from the script:
+
+```bash
+--set controller.podLabels.aadpodidbinding=$AAD_POD_IDENTITY_NAME \
```
Comment (aramase) [inline docs/ingress-tls.md:154]: this is still required if not using service principal mode

File: docs/ingress-tls.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+Next, [Deploy the application](#Deploy-Application-with-Ingress-reference-to-Secrets-Store-CSI).
+
+## Deploy Test Apps
+
+Depending on the TLS certificate lifecycle, follow one of the following steps:
+
+* ### Deploy Application with Reference to Secrets Store CSI
+
+> NOTE: These apps reference a Secrets Store CSI volume and a `secretProviderClass` object created earlier. A Kubernetes secret `ingress-tls-csi` will be created by the CSI driver as a result of the app creation.
```
Comment (aramase) [inline docs/ingress-tls.md:179]:
```suggestion
> NOTE: These apps reference a Secrets Store CSI volume and a `secretProviderClass` object created earlier. A Kubernetes secret `ingress-tls-csi` will be created by the CSI driver as a result of the app creation in the same namespace.
```

File: docs/ingress-tls.md (hunk 1 above)
Comment (aramase) [reply docs/ingress-tls.md:8]: If the ingress and related certificates are managed by the IT Unit/cluster admin does this mean they're also responsible for creating and managing the ingress resources?

**Review by aramase — COMMENTED** [review]
Thank you for the PR!

1. Can we keep the dir name as `sample` instead of changing to `samples`?
2. Could we also add a readme in the dir to reference to the docs? The CSI driver repo adds a reference link to the docs in the repo - https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/master/sample/ingress-controller-tls

File: docs/ingress-tls.md (hunk 1 above)
Comment (balteravishay) [reply docs/ingress-tls.md]: 1. yes, this is something some customers do, and is documented. for instance here in the nginx ingress docs: https://docs.nginx.com/nginx-ingress-controller/installation/running-multiple-ingress-controllers/#running-multiple-nginx-ingress-controllers and here in the ms-pnp AKS security best practices guidelines: https://github.com/mspnp/aks-secure-baseline/blob/main/workload/traefik.yaml
2. added a note that in this scenario the ingress is namespace/workload specific.

File: docs/ingress-tls.md (hunk 2 above)
Comment (balteravishay) [reply docs/ingress-tls.md]: removed password from script.

File: docs/ingress-tls.md (hunk 4 above)
Comment (balteravishay) [reply docs/ingress-tls.md]: fixed

**Comment by balteravishay** [comment]
@aramase i kept the dir as sample, and added a readme. not sure if the readme content is as you requested.
other comments are resolved. 
Thanks!

**Review by aramase — APPROVED** [review]
lgtm

The URL [here](https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/master/sample/ingress-controller-tls) needs to be updated after the PR is merged.

File: docs/ingress-tls.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+### Add the official ingress chart repository
+
+```bash
+helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
+helm repo update
+```
+
+### Deploy Ingress
+
+Depending on the TLS certificate lifecycle, follow one of the following steps:
```
Comment (ritazh) [inline docs/ingress-tls.md:105]: Thanks for the PR @balteravishay! Can you please add few comments around when a user would choose to add the CSI volume to the ingress controller deployment and when they would choose to add it an application deployment?

Comment (ritazh) [reply docs/ingress-tls.md:105]: Resolved as it has been address [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/pull/282/files#diff-2da0ddc89b1dfa84a2cff2521f42abfbafb87540a6bd531f2ca1cdd32ad3aed6R6-R9)

**Review by ritazh — APPROVED** [review]

**Pull request merged**